test/performance/shmem_perf_suite/common.h

```diff
@@ -47,6 +47,148 @@
 
 #define ONE 1
 
+/* constants for experiments */
+#define MAX_MSG_SIZE (1<<23)
+#define START_LEN 1
+#define INC 2
+#define TRIALS 500
+#define WINDOW_SIZE 64
+#define WARMUP 50
+
+/* constants for experiments with large message sizes */
+#define TRIALS_LARGE  100
+#define WINDOW_SIZE_LARGE 64
+#define WARMUP_LARGE  10
+#define LARGE_MESSAGE_SIZE  8192
+
+#define TARGET_SZ_MIN 8
+#define TARGET_SZ_MAX 4096
+
+/* atomics common */
+#define ATOMICS_N_DTs 3
+/* note: ignoring cswap/swap for now in verification */
+#define ATOMICS_N_OPs 4
+/* PE 0 is printing its latency, thus have it not be the INCAST PE*/
+#define INCAST_PE 1
+
+/* perf metrics structures */
+typedef enum {
+    LAT,
+    BW
+} test_type;
+
+typedef enum {
+    UNI_DIR,
+    BI_DIR
+} bw_type;
+
+typedef enum {
+    STYLE_PUT,
+    STYLE_GET,
+    STYLE_RMA,
+    STYLE_ATOMIC
+} bw_style;
+
+typedef enum {
+    FIRST_HALF,
+    SECOND_HALF,
+    FULL_SET
+} red_PE_set;
+
+typedef enum {
+    COMM_PAIRWISE,
+    COMM_INCAST
+} comm_style;
+
+typedef enum {
+    B,
+    KB,
+    MB
```

Comment:
Should these have specific values, e.g. B=1, KB=1024, etc?

Reply:
This ``enumerator`` type is used to just detect how the output bandwidth is to be reported. Based on the enum values, we multiply with the appropriate value. We can go either way for this.

Reply:
@jdinan, Keeping this as it is for now. Will refine this in a future PR.